src/main/java/net/neoforged/neoforge/common/IOUtilities.java

```diff
@@ -44,14 +46,18 @@ private IOUtilities() {}
      * @param prefix     The prefix of temporary files to clean up, or null if all
      *                   temporary files should be removed.
      *
-     * @throws IOException if an I/O error occurs during deletion.
+     * @implNote IOException is wrapped to RuntimeException, which will be logged to the error log by the ioPool.
      */
-    public static void cleanupTempFiles(Path targetPath, @Nullable String prefix) throws IOException {
-        try (var filesToDelete = Files.find(targetPath, 1, createPredicate(prefix))) {
-            for (var file : filesToDelete.toList()) {
-                Files.deleteIfExists(file);
+    public static CompletableFuture<?> cleanupTempFiles(CompletableFuture<?> pendingWriteFuture, Path targetPath, @Nullable String prefix) {
+        return pendingWriteFuture.thenRunAsync(() -> {
+            try (var filesToDelete = Files.find(targetPath, 1, createPredicate(prefix))) {
+                for (var file : filesToDelete.toList()) {
+                    Files.deleteIfExists(file);
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
I'm note sure if we want to be throwing an error here as it could get eaten by the thread. It would probably be better to log the exception and report it as an error in the console.

Reply:
that is already handled in `net.minecraft.Util#onThreadException` via `net.minecraft.Util#makeIoExecutor`

Reply:
Fair enough, as long as the exception reports the saved data info.